cupy/padding/pad.py

```diff
@@ -1,321 +1,786 @@
-                             narray), axis=axis)
+###############################################################################
+# Private utility functions.
 
 
-def _append_const(narray, pad_amount, value, axis=-1):
-    if pad_amount == 0:
-        return narray
-    padshape = tuple([x if i != axis else pad_amount
-                      for i, x in enumerate(narray.shape)])
-    return cupy.concatenate((narray,
-                             cupy.full(padshape, value, narray.dtype)),
-                            axis=axis)
+def _linear_ramp(ndim, axis, start, stop, size, reverse=False):
+    """Create a linear ramp of `size` in `axis` with `ndim`.
 
+    This algorithm behaves like a vectorized version of `numpy.linspace`.
+    The resulting linear ramp is broadcastable to any array that matches the
+    ramp in `shape[axis]` and `ndim`.
 
-def _prepend_edge(arr, pad_amt, axis=-1):
-    """Prepend `pad_amt` to `arr` along `axis` by extending edge values.
+    Args:
+      ndim(int): Number of dimensions of the resulting array. All dimensions
+          except the one specified by `axis` will have the size 1.
+      axis(int): The dimension that contains the linear ramp of `size`.
+      start(int or ndarray): The starting value(s) of the linear ramp. If given
+          as an array, its size must match `size`.
+      stop(int or ndarray): The stop value(s) (not included!) of the linear
+          ramp. If given as an array, its size must match `size`.
+      size(int): The number of elements in the linear ramp. If this argument is
+          0 the dimensions of `ramp` will all be of length 1 except for the one
+          given by `axis` which will be 0.
+      reverse(bool, optional): If False, increment in a positive fashion,
+          otherwise decrement. (Default value = False)
 
-    Parameters
-    ----------
-    arr : ndarray
-        Input array of arbitrary shape.
-    pad_amt : int
-        Amount of padding to prepend.
-    axis : int
-        Axis along which to pad `arr`.
+    Returns:
+      ndarray: Output array of dtype cupy.float64 that in- or decrements along
+      the given `axis`.
+
+    Examples
+    --------
+    >>> _linear_ramp(ndim=2, axis=0, start=cupy.arange(3), stop=10, size=2)
+    array([[0. , 1. , 2. ],
+           [5. , 5.5, 6. ]])
+    >>> _linear_ramp(ndim=3, axis=0, start=2, stop=0, size=0)
+    array([], shape=(0, 1, 1), dtype=float64)
+    """
+    # Create initial ramp
+    ramp = cupy.arange(size, dtype=cupy.float64)
+    if reverse:
+        ramp = ramp[::-1]
 
-    Returns
-    -------
-    padarr : ndarray
-        Output array, extended by `pad_amt` edge values appended along `axis`.
+    # Make sure, that ramp is broadcastable
+    init_shape = (1,) * axis + (size,) + (1,) * (ndim - axis - 1)
+    ramp = ramp.reshape(init_shape)
 
+    if size != 0:
+        # And scale to given start and stop values
+        gain = (stop - start) / float(size)
```

Comment:
yeah, this was listed as a TODO for the first comment. I see that those PRs have been merged now, so I will make the corresponding changes here.

Reply:
I added the changes from numpy/numpy#14209, but had to fall back to numpy's linspace since the `axis` argument is not implemented in `cupy.linspace`. That can probably be done, but should be a separate PR.